employee/read_only.c

```diff
@@ -6,23 +6,28 @@
 
 #include "../include/common.h"
 #include "include/employee.h"
+#include <assert.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-int main()
+int main(int argc, char *argv[])
 {
-	struct employee employee = {0, "Good", "Employee", 55000};
+	struct employee employee = {1000, "Good", "Employee", 55000.0};
 	double salary = 0.0;
+	printf("--- EMPLOYEE ---\n");
 	// set the permissions to read-only
 	struct employee *ro_employee = set_read_only(&employee);
-	inspect_pointer(ro_employee);
+	assert((cheri_perms_get(ro_employee) & CHERI_PERM_STORE) == 0);
 	print_details(ro_employee);
-	printf("The struct is read-only so trying to change the salary will make the program crash...");
+	printf("\nThe struct is read-only so trying to change the salary will make the program "
+		   "crash...\n");
 	printf("\nInsert the new salary for this employee: \n");
 	fflush(stdout);
-	scanf("%lf", &salary);
-
+	if (argc > 1)
+		salary = 65000.0;
```

Comment:
Should this case read the salary from `argv[1]`?

Reply:
Well the `scanf` approach used below would make more sense than `atoi`. Really, though, it just looked like a placeholder value.

The example would work fine if it didn't prompt (just picking some arbitrary new value), and then you can remove the argument parsing code entirely. Simplifying boilerplate out of examples is usually a good idea.

Reply:
This looks an excellent suggestion. I'll fix it as you have advised.

Reply:
Fixed in e1b25f1.